src/Microsoft.Health.Dicom.Api/Controllers/WorkitemController.Add.cs

```diff
@@ -46,23 +49,23 @@ public partial class WorkitemController
     [VersionedPartitionRoute(KnownRoutes.AddWorkitemInstancesRoute, Name = KnownRouteNames.PartitionedAddWorkitemInstance)]
     [VersionedRoute(KnownRoutes.AddWorkitemInstancesRoute, Name = KnownRouteNames.AddWorkitemInstance)]
     [AuditEventType(AuditEventSubType.AddWorkitem)]
-    public async Task<IActionResult> AddAsync()
+    public async Task<IActionResult> AddAsync([FromBody][Required][MinLength(1)][MaxLength(1)] IReadOnlyCollection<DicomDataset> dicomDatasets)
     {
         // The Workitem UID is passed as the name of the first query parameter 
         string workitemUid = HttpContext.Request.Query.Keys.FirstOrDefault();
 
-        return await PostAddAsync(workitemUid);
+        return await PostAddAsync(workitemUid, dicomDatasets);
     }
 
-    private async Task<IActionResult> PostAddAsync(string workitemInstanceUid)
+    private async Task<IActionResult> PostAddAsync(string workitemInstanceUid, IReadOnlyCollection<DicomDataset> dicomDatasets)
     {
-        long fileSize = Request.ContentLength ?? 0;
-        _logger.LogInformation("DICOM Web Add Workitem Transaction request received, with Workitem instance UID {WorkitemInstanceUid}, and file size of {FileSize} bytes",
-            workitemInstanceUid ?? string.Empty,
-            fileSize);
+        long numberOfDatasets = dicomDatasets?.Count ?? 0;
```

Comment:
The length is guaranteed to be 1 and there's no need to `null` check either, right? We are validating this now